### ISSUE TITLE
Bucket doesn't exist after expanding distributed cluster

### ISSUE THREAD

**Issue by weironz** [minio/minio#11209]
this bug still exits ,but why closed 

[10599](https://github.com/minio/minio/issues/10599)

what version i use:
```
[root@minio-1 /]# minio --version
minio version RELEASE.2020-12-29T23-29-29Z
```

after expanding my cluster, then up load files to exits bucket error logs
```
[root@minio-1 ~]# mc ls minio
[2021-01-03 16:41:11 CST]     0B asdfasdf/
[2021-01-03 16:46:22 CST]     0B new112/
[root@minio-1 ~]# mc cp * minio/asdfasdf/
mc: <ERROR> Failed to copy `anaconda-ks.cfg`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `789`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `currentbuild.zip`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `bad.yaml`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `docker-compose-secrets.yaml`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `nohup.out`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `minio:test`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `rclone-v1.53.3-linux-amd64.zip`. Bucket `asdfasdf` does not exist.
mc: <ERROR> Failed to copy `rclone-v1.53.3-linux-amd64.zip`. Bucket `asdfasdf` does not exist.
rclone-v1.53.3-linux-amd64.zip:      584.94 KiB / 29.26 MiB ┃▓▓█░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
``

**Comment by harshavardhana** [comment]
@willzhang can you provide a reproducer for this? - Ic can't seem to reproduce this

**Comment by weironz** [comment]
> @willzhang can you provide a reproducer for this? - Ic can't seem to reproduce this

## reproducer:

prepare 4 nodes, node1 node2 node3 node4, and create mount point
```
[root@minio-1 ~]# df -h |grep data
/dev/mapper/vg01-lv01     20G   33M   20G   1% /data/minio
```

create a new cluster, run follow command in node1 and node2
```
docker run -d --name minio \
  --restart=always --net=host \
  -e MINIO_ACCESS_KEY=minio \
  -e MINIO_SECRET_KEY=minio123 \
  -v /data/minio/data1:/data1 \
  -v /data/minio/data2:/data2 \
  minio/minio server \
  http://192.168.92.{10...11}/data{1...2}
```

create a bucket named asdfasdf and upload some date to it，then stop node1 node2 minio service:

```
docker stop minio && docker rm minio
```

expand cluster ,add node3 and node4,  run follow commands in all 4 nodes 

```
docker run -d --name minio \
  --restart=always \
  --net=host \
  -e MINIO_ACCESS_KEY=minio \
  -e MINIO_SECRET_KEY=minio123 \
  -v /data/minio/data1:/data1 \
  -v /data/minio/data2:/data2 \
  minio/minio server \
  http://192.168.92.{10...11}/data{1...2} \
  http://192.168.92.{12...13}/data{1...2}
```

now upload date to asdfasdf buckets will failed
```
Bucket `asdfasdf` does not exist.
```

and the minio logs

```
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:15 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
[16:49:33]
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:32 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
[16:49:33]
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:32 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
[16:49:33]
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:32 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
[16:49:33]
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:32 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
[16:49:33]
[16:49:33]API: SYSTEM()
[16:49:33]Time: 08:46:32 UTC 01/03/2021
[16:49:33]DeploymentID: 36be5caa-f3d4-42e3-9008-23fba03df77a
[16:49:33]Error: volume not found (*errors.errorString)
[16:49:33]       1: cmd/metacache-set.go:844:cmd.listPathRaw.func1()
```
when i use minio/minio:RELEASE.2020-09-23T19-18-30Z  which tags is before 2020-09-26 have no problems as the [10599](https://github.com/minio/minio/issues/10599#issuecomment-701085820) issue say.

**Comment by harshavardhana** [comment]
To be fixed by #11207

**Comment by weironz** [comment]
not fixed, still have problems, reproducer  see https://github.com/minio/minio/issues/11209#issuecomment-753599268

and error logs different ,as follow

```
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
Waiting for all MinIO sub-systems to be initialized.. lock acquired
Verifying if 1 bucket is consistent across drives...
Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
```

**Comment by harshavardhana** [comment]
> not fixed, still have problems, reproducer see [#11209 (comment)](https://github.com/minio/minio/issues/11209#issuecomment-753599268)
> 
> and error logs different ,as follow
> 
> ```
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> Waiting for all MinIO sub-systems to be initialized.. lock acquired
> Verifying if 1 bucket is consistent across drives...
> Waiting for all MinIO sub-systems to be initialized.. possible cause (Unable to list buckets to heal: Storage resources are insufficient for the write operation demo123/)
> ```

You need all servers online to heal the cluster @willzhang
